handler/core/src/main/java/io/knotx/fragments/task/TaskProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.task;
+
+import io.knotx.fragments.engine.FragmentEventContext;
+import io.knotx.fragments.engine.FragmentsEngine;
+import io.knotx.fragments.engine.Task;
+
+/**
+ * Produces {@link Task} based on graph configuration, Fragment data and request.
+ */
+public interface TaskProvider {
```

Comment:
Well - it more or less looks to me like a Factory. But `provider` is fine name also.

Reply:
I tried to prevent `TaskFactoryFactory`.